suite-native/module-onboarding/src/screens/OnboardingXpubScan.tsx

```diff
@@ -0,0 +1,131 @@
+import React, { useCallback, useState } from 'react';
+import { Pressable, View } from 'react-native';
+
+import { useFocusEffect } from '@react-navigation/native';
+
+import { Screen, StackProps } from '@suite-native/navigation';
+import { Box, Chip, Input, InputWrapper, Text } from '@suite-native/atoms';
+import { prepareNativeStyle, useNativeStyles } from '@trezor/styles';
+import { CryptoIcon } from '@trezor/icons';
+import { NetworkKey } from '@suite-common/wallet-config';
+
+import { Camera, CAMERA_HEIGHT } from '../components/Camera';
+import { OnboardingStackParamList, OnboardingStackRoutes } from '../navigation/routes';
+
+const xpubContainerStyle = prepareNativeStyle(utils => ({
+    padding: utils.spacings.medium,
+}));
+
+const coinStyle = prepareNativeStyle(_ => ({
+    flexDirection: 'row',
+}));
+
+const cameraStyle = prepareNativeStyle(_ => ({
+    marginTop: 20,
+    marginBottom: 45,
+}));
+
+const cameraPlaceholderStyle = prepareNativeStyle(utils => ({
+    height: CAMERA_HEIGHT,
+    borderRadius: utils.borders.radii.medium,
+    alignItems: 'center',
+    justifyContent: 'center',
+    backgroundColor: utils.colors.gray800,
+}));
+
+const chipStyle = prepareNativeStyle(utils => ({
+    flex: 1,
+    borderRadius: utils.borders.radii.small,
+}));
+
+const DEFAULT_XPUB_INPUT_TEXT = '';
+const DEFAULT_CURRENCY_SYMBOL = 'btc';
+
+export const OnboardingXpubScan = ({
+    navigation,
+}: StackProps<OnboardingStackParamList, OnboardingStackRoutes.OnboardingXpubScan>) => {
+    const [selectedCurrencySymbol, setSelectedCurrencySymbol] =
+        useState<NetworkKey>(DEFAULT_CURRENCY_SYMBOL);
+    const [inputText, setInputText] = useState<string>(DEFAULT_XPUB_INPUT_TEXT);
+    const [cameraRequested, setCameraRequested] = useState<boolean>(false);
+    const { applyStyle } = useNativeStyles();
+
+    useFocusEffect(
```

Comment:
Using hook in hook is not good idea, you should create some handler `resetToDefaultValues` with useCallback before and then only use it here.